docs/admin/guides/build-your-own-pulp-image.md

```diff
@@ -0,0 +1,147 @@
+# Build your own Pulp image
+
+The Pulp images (`pulp/pulp` and `pulp/pulp-minimal`) come pre-installed with the set of plugins
+we officially support. However, these images can be easily customized to suit your specific content
+needs. 
+
+## Change the versions of the plugins
+
+The easiest way to customize the images is to specify the plugin versions that should be used when
+building them. Clone and check out the `pulp-oci-images` repository to use the following commands.
+
+```bash
+# Using docker to build pulp/pulp with a custom pulpcore version
+$ docker build --build-arg PULPCORE_VERSION="==3.65.0" --file images/pulp/Containerfile --tag pulp/pulp:custom
+# Using buildah to build pulp/pulp-minimal & pulp-web with a custom pulp_gem version
+$ buildah bud --build-arg PULP_GEM_VERSION=">=2.22.0" --file images/pulp-minimal/Containerfile.core --tag pulp/pulp-minimal:custom
+$ buildah bud --build-arg FROM_TAG="custom" --file images/pulp-minimal/Containerfile.webserver --tag pulp/pulp-web:custom
+```
+
+## Add/remove plugins
+
+If you want to build a custom image with a different set of plugins then the official images you are
+going to need to write a custom Containerfile. The Containerfiles for `pulp` and `pulp-minimal` 
+should be used as templates for creating your custom image. First clone and check out the 
+`pulp-oci-images` repository. Then create a new directory in the `images/` directory for your 
+custom image and copy over the corresponding Containerfile for the `pulp` or `pulp-minimal` image
+into your new directory. See [Available Images](site:pulp-oci-images/docs/admin/reference/available-images/)
+for more information on the difference between the two images.
+
+### pulp image customization
+
+Here is an example of a customized Containerfile for the multiprocess image.
+
+```dockerfile title="custom/Containerfile"
+ARG FROM_TAG="latest"
+# The multiprocess pulp image must inherit from pulp/pulp-ci-centos9
+# The tags follow pulpcore-versioning 3.Y(.Z), we recommend using "latest"
+FROM pulp/pulp-ci-centos9:${FROM_TAG}
+
+# These are extra pip requirements needed to run Pulp
+COPY images/assets/requirements.extra.txt /requirements.extra.txt
+
+# Here you customize the plugins and their versions that you want to install
+RUN pip3 install --upgrade \
+  pulpcore[s3,google,azure] \
+  pulp-custom-plugin \
+  -r /requirements.extra.txt \
+  -c /constraints.txt && \
+  rm -rf /root/.cache/pip
+
+# collectstatic makes the api browsable in a web browser
+USER pulp:pulp
+RUN PULP_STATIC_ROOT=/var/lib/operator/static/ PULP_CONTENT_ORIGIN=localhost \
+       /usr/local/bin/pulpcore-manager collectstatic --clear --noinput --link
+USER root:root
```

Comment:
Unrelated question:
So we leave root as the last user here. Does this mean we leave dropping the privileges to the s6 scripts later?

Reply:
Yes, they have the line `s6-setuidgid pulp`, in their run description.